presto-docs/src/main/sphinx/security/server.rst

```diff
@@ -0,0 +1,281 @@
+  What is your first and last name?
+    [Unknown]:  presto-coordinator
+  What is the name of your organizational unit?
+    [Unknown]:
+  What is the name of your organization?
+    [Unknown]:
+  What is the name of your City or Locality?
+    [Unknown]:
+  What is the name of your State or Province?
+    [Unknown]:
+  What is the two-letter country code for this unit?
+    [Unknown]:
+  Is CN=eiger, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown correct?
+    [no]:  yes
+
+  Enter key password for <presto>
+          (RETURN if same as keystore password):
+
+
+.. _server_access_controller:
+
+Access Controller Implementation
+--------------------------------
+
+Presto separates the concept of the principal who authenticates to the
+coordinator from the username that is responsible for running queries. When
+running the Presto CLI, for example, the Presto username can be specified using
+the --user option.
+
+By default, the Presto coordinator allows any principal to run queries as any
+Presto user. In a secure environment, this is probably not desirable behavior.
+
+This behavior can be customized by implementing the
+``SystemAccessControlFactory`` and ``SystemAccessControl`` interfaces.
+
+``SystemAccessControlFactory`` is responsible for creating a
+``SystemAccessControl`` instance. It also defines a ``SystemAccessControl``
+name which is used by the administrator in a Presto configuration.
+
+``SystemAccessControl`` implementations have two responsibilities:
+ * Verifying whether or not a given Kerberos principal is authorized to execute
+   queries as a specific user.
+ * Determining whether or not a given user can alter values for a given system
+   property.
+
+The implementation of ``SystemAccessControl`` and
+``SystemAccessControlFactory`` must be wrapped as a plugin and installed on the
+Presto cluster.
+
+Presto Coordinator Node Configuration
+-------------------------------------
+
+You must make the above changes to the environment prior to configuring the
+Presto coordinator to use Kerberos authentication and HTTPS. After making the
+following environment changes, you can make the changes to the Presto
+configuration files.
+
+ * :ref:`server_kerberos_services`
+ * :ref:`server_kerberos_configuration`
+ * :ref:`server_kerberos_principals`
+ * :ref:`server_java_keystore`
+ * :ref:`server_access_controller`
+
+config.properties
+^^^^^^^^^^^^^^^^^
+
+Kerberos authentication is configured in the coordinator node's
+config.properties file. The entries that need to be added are listed below.
+
+.. code-block:: none
+
+  http.server.authentication.enabled=true
+
+  http.server.authentication.krb5.service-name=presto
+  http.server.authentication.krb5.keytab=/etc/presto/presto.keytab
+  http.authentication.krb5.config=/etc/krb5.conf
+
+  http-server.https.enabled=true
+  http-server.https.port=7778
+
+  http-server.https.keystore.path=/etc/presto_keystore.jks
+  http-server.https.keystore.key=keystore_password
```

Comment:
We should also add a recommendation to limit the cipher suite.  By default Java will choose the most expensive cipher suite and this can cause the coordinator to use 90+%  of all cpu doing HTTPS.  We have found the following to be the most secure suites that run efficiently in Java:
```
http-server.https.included-cipher=TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA256
```